frontend/src/components/managePermissions/table/ApplicationAdminTable.vue

```diff
@@ -0,0 +1,169 @@
+<script setup lang="ts">
+import { reactive, ref, type PropType  } from 'vue';
+import { FilterMatchMode } from 'primevue/api';
+import Column from 'primevue/column';
+import DataTable from 'primevue/datatable';
+import { useConfirm } from 'primevue/useconfirm';
+import ConfirmDialog from 'primevue/confirmdialog';
+
+import { IconSize } from '@/enum/IconEnum';
+import { routeItems } from '@/router/routeItem';
+import Button from '@/components/common/Button.vue';
+import ConfirmDialogtext from '@/components/managePermissions/ConfirmDialogText.vue';
+import DataTableHeader from '@/components/managePermissions/table/DataTableHeader.vue';
+import type { FamAppAdminGet } from 'fam-admin-mgmt-api/model';
+
+type emit = (
+    e: 'deleteAppAdmin',
+    item: FamAppAdminGet
+) => void;
+
+const props = defineProps({
+    loading: {
+        type: Boolean,
+        default: false,
+    },
+    applicationAdmins: {
+        type: [Array] as PropType<
+            FamAppAdminGet[] | undefined
+        >,
+        required: true,
+    }
+});
+
+const adminFilters = ref({
+    global: { value: '', matchMode: FilterMatchMode.CONTAINS },
+    'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
+    'user.user_type.description': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'application.app_environment': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+
+    },
+});
+
+const confirm = useConfirm();
+
+const emit = defineEmits<emit>();
+
+const confirmDeleteData = reactive({
+    adminName: '',
+    role: 'ADMIN'
+});
+
+const adminSearchChange = (newvalue: string) => {
+    adminFilters.value.global.value = newvalue;
+};
+
+const deleteAdmin = (admin: FamAppAdminGet) => {
+    confirmDeleteData.adminName = admin.user.user_name;
+    confirm.require({
+        group: 'deleteAdmin',
+        header: 'Remove Access',
+        rejectLabel: 'Cancel',
+        acceptLabel: 'Remove',
+        acceptClass: 'p-button-danger',
+        accept: () => {
+            emit('deleteAppAdmin', admin);
+        },
+    });
+}
+
+</script>
+
+<template>
+    <ConfirmDialog group="deleteAdmin">
+        <template #message>
+            <ConfirmDialogtext
+                :userName="confirmDeleteData.adminName"
+                :role="confirmDeleteData.role"
+            />
+        </template>
+    </ConfirmDialog>
+    <div class="data-table-container">
+        <div class="custom-data-table">
+            <DataTableHeader
+                btnLabel="Add application admin"
+                :btnRoute="routeItems.grantUserPermission.path"
```

Comment:
just a reminder to Nick, this has change to the new router for granting application admin